src/Security/PermissionManager.php

```diff
@@ -0,0 +1,265 @@
+                    'AND' => [
+                        Profile_User::getTableField('entities_id') => $parent_entities,
+                        Profile_User::getTableField('is_recursive') => 1
+                    ]
+                ]
+            ]
+        ]);
+        foreach ($iterator as $row) {
+            $profiles[] = $row['profiles_id'];
+        }
+
+        return $profiles;
+    }
+
+    /**
+     * Combines the permissions granted by all the provided profiles.
+     *
+     * @param array $profiles Array of profile IDs.
+     * @return array Array of rights combined from all provided profiles
+     */
+    public static function getAggregatedRights(array $profiles): array
+    {
+        global $DB;
+
+        if (empty($profiles)) {
+            return [];
+        }
+
+        $rights = [];
+
+        $iterator = $DB->request([
+            'SELECT' => ['name', 'rights'],
+            'FROM'   => ProfileRight::getTable(),
+            'WHERE'  => [
+                ProfileRight::getTableField('profiles_id') => $profiles
+            ]
+        ]);
+
+        foreach ($iterator as $row) {
+            $right_name = $row['name'];
+            if (!isset($rights[$right_name])) {
+                $rights[$right_name] = 0;
+            }
+            $rights[$right_name] |= $row['rights'];
+        }
+
+        return $rights;
+    }
+
+    /**
+     * @param int $users_id The user's ID or -1 for the current user.
+     *                      While you could pass the actual ID of the current user, it is not recommended as it would
+     *                      bypass optimizations that are made by using data already available in the session.
+     * @param string $module The module to check. Typically, this is a $rightname property of a class.
+     * @param int $right The individual right/permission to check. Common values are defined by the constants:
+     *                   {@link READ}, {@link UPDATE}, {@link CREATE}, {@link DELETE}, {@link PURGE}, {@link PURGE},
+     *                   {@link ALLSTANDARDRIGHT}, {@link READNOTE}, {@link UPDATENOTE}, and {@link UNLOCK}.
+     *                   Rights unique to a module may be defined as constants within their related class.
+     *                   Not all the common rights are valid for all modules. For example, an item that cannot be soft-deleted
+     *                   would only have the {@link PURGE} right and not the {@link DELETE} right.
+     * @param bool $all_profiles If this is true or if the provided user ID is not -1, every profile assigned to the user will be checked.
+     *                           Otherwise, if the users_id is -1 and this is false, only the active profile is checked.
+     * @param int $entities_id The entity to check. If -1 provided, the current entity is used.
+     *
+     * @return bool
+     */
+    public static function haveRight(int $users_id, string $module, int $right, bool $all_profiles = false, int $entities_id = -1): bool
```

Comment:
Moving the `$user_id` in 4rd position would propably prevent people to pass a value in this argument when checking the current user in session.

```suggestion
    public static function haveRight(string $module, int $right, bool $all_profiles = false, int $users_id = -1, int $entities_id = -1): bool
```